litecommands-fabric/src/main/java/dev/rollczi/litecommands/fabric/FabricScheduler.java

```diff
@@ -0,0 +1,50 @@
+package dev.rollczi.litecommands.fabric;
+
+import dev.rollczi.litecommands.scheduler.AbstractMainThreadBasedScheduler;
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import net.minecraft.util.thread.ReentrantThreadExecutor;
+
+public abstract class FabricScheduler<R extends Runnable> extends AbstractMainThreadBasedScheduler {
+
+    private final ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+    private final ExecutorService asyncExecutor;
+
+    protected FabricScheduler() {
+        this.asyncExecutor = Executors.newFixedThreadPool(Math.max(1, Runtime.getRuntime().availableProcessors() / 2));
+    }
+
+    protected FabricScheduler(ExecutorService asyncExecutor) {
+        this.asyncExecutor = asyncExecutor;
+    }
+
+    @Override
+    protected void runSynchronous(Runnable task, Duration delay) {
+        if (delay.isZero()) {
+            this.getMainThreadExecutor().submit(task);
+        } else {
+            this.scheduledExecutor.schedule(() -> this.getMainThreadExecutor().submit(task), delay.toMillis(), TimeUnit.MILLISECONDS);
+        }
+    }
+
+    @Override
+    protected void runAsynchronous(Runnable task, Duration delay) {
+        if (delay.isZero()) {
+            this.getMainThreadExecutor().submit(task);
+        } else {
+            this.scheduledExecutor.schedule(() -> this.getMainThreadExecutor().submit(task), delay.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
This will cause tasks to be executed on the main game thread, and the meaning of asynchronousness will be lost.
```suggestion
            this.asyncExecutor.schedule(task, delay.toMillis(), TimeUnit.MILLISECONDS);
```

Reply:
Thanks  for your review @huanmeng-qwq  I missed that